extensions/amp-ima-video/0.1/amp-ima-video.js

```diff
@@ -64,16 +64,29 @@ class AmpImaVideo extends AMP.BaseElement {
     assertHttpsUrl(this.element.getAttribute('data-tag'),
         'The data-tag attribute is required for <amp-video-ima> and must be ' +
             'https');
-    assertHttpsUrl(this.element.getAttribute('data-src'),
-        'The data-src attribute is required for <amp-video-ima> and must be ' +
-            'https');
+
+    // Set data-sources attribute based on source child elements.
+    const sourceElements = this.element.getElementsByTagName('source');
+    if (sourceElements.length > 0) {
+      const sources = [];
+      Array.from(sourceElements).forEach(source => {
+        sources.push({
+          'src': source.src,
```

Comment:
Ah ok, I misunderstood your comment in the initial bug - when you were talking about "adding source nodes later" causing problems, I thought you meant programmatically adding them to the video player. Fixed now.